pandas/core/internals/array_manager.py

```diff
@@ -869,6 +869,25 @@ def iset(
             self.arrays[mgr_idx] = value_arr
         return
 
+    def column_setitem(
+        self, loc: int, idx: int | slice | np.ndarray, value, inplace: bool = False
+    ) -> None:
+        """
+        Set values ("setitem") into a single column (not setting the full column).
+
+        This is a method on the ArrayManager level, to avoid creating an
+        intermediate Series at the DataFrame level (`s = df[loc]; s[idx] = value`)
+        """
+        arr = self.arrays[loc]
+        # create temporary SingleArrayManager without ref to use setitem implementation
+        mgr = SingleArrayManager([arr], [self._axes[0]])
+        if inplace:
+            mgr.setitem_inplace(idx, value)
+        else:
+            new_mgr = mgr.setitem((idx,), value)
+            # update existing ArrayManager in-place
+            self.arrays[loc] = new_mgr.arrays[0]
```

Comment:
could we share code by using ArrayManager.iset here?  (though it looks like that method ignores the 'inplace' kwd)

Reply:
I would prefer not to, because I will then have to add this again anyway in https://github.com/pandas-dev/pandas/pull/46958/ (since that PR is only doing the CoW handling for BlockManager), limiting the goal of reducing the diff in that PR. 

Now, I could indeed use `iset` here, but that does some additional (unnecessary in this case) validation, so _if_ we have a separate implementation like the one above, I think it's fine to use `self.arrays[loc] = new_mgr.arrays[0]` instead of `iset` (both are a single line of code, and the former is the essence of what `iset` is doing but then without all extra checks)